File: 13_NoSQL/Uebungsdb Exams/Model/Term.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace ExamDbGenerator.Model
{
    record Term(int Year, [property: BsonRepresentation(BsonType.String)] TermType TermType, DateOnly Start, DateOnly End)
    {
        [BsonElement]
        public string Id => TermType == TermType.Year ? Year.ToString() : $"{Year}{TermType.ToString()[0]}";
    }

}

File: 13_NoSQL/examsdb-client-java/src/main/java/at/spengergasse/model/Term.java
package at.spengergasse.model;

import lombok.*;
import org.bson.codecs.pojo.annotations.BsonId;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Term {
    private int year;
    private TermType termType;
    private DateOnly start;
    private DateOnly end;

    @BsonId
    public String getId() {
        return String.format("%s%s", String.valueOf(year), termType.toString().charAt(0));
    }
}
